tests/unit/test_security_middleware_rate_limit.py

```
@@ -0,0 +1,283 @@
+        from utils.rate_limit import TokenBucket
+
+        # Replace bucket with a zero-refill one for determinism
+        mw._mutating_bucket = TokenBucket(capacity=10, refill_rate=0)
+        client = middleware_app.test_client()
+        for _ in range(10):
+            resp = client.post("/update_now")
+            assert resp.status_code == 200
+        resp = client.post("/update_now")
+        assert resp.status_code == 429
+
+    def test_save_plugin_settings_burst_then_429(self, middleware_app, monkeypatch):
```

Ruff (0.15.9)

[warning] Unused method argument: `monkeypatch`

(ARG002)